### ISSUE TITLE
输出tensor和原onnxruntime输出不符合

### ISSUE THREAD

**Issue by lqian** [alibaba/MNN#844]
运行和实验环境: Ubuntu 16.05 LTS, MNN 1.0.0, conda3, python 3.6, pytorch 1.2.0,  onnxruntime 1.3.0

实验步骤：
1, 从pytorch mobilenetv2骨干网中取base.feature_1部分, 组建一个独立的网络，testPartModule.pth，导出onnx模型，testPartModule.onnx；比较pytroch中运行base_feature1网络输出和onnxruntime的推理输出，结果一致
2. 把testPartModule.onnx转换成MNN模型，testPartModule.mnn，测试比较发现MNN推理的结果和onnx推理结果不相符。

------ 测试代码和模型 ---------
https://pan.baidu.com/s/1ZIo9Jvel_3hAQaoaDs7-yA 
提取码: 6xqg 

--- pytorch 输出 ---
torch.Size([1, 24, 96, 128])
tensor([1.3382, 1.0491, 0.5778, 0.5709, 0.5709, 0.5709, 0.5709, 0.5709, 0.5709,
        0.5709, 0.5709, 0.5709, 0.5709, 0.5709, 0.5709, 0.5709, 0.5709, 0.5709,
        0.5709, 0.5709])

--- onnx runtime 输出 ----
[1.3382119 1.0490794 0.577814  0.5708798 0.5708798 0.5708798 0.5708798
 0.5708798 0.5708798 0.5708798 0.5708798 0.5708798 0.5708798 0.5708798
 0.5708798 0.5708798 0.5708798 0.5708798 0.5708798 0.5708798]

---  MNN 输出 ----
output tensor shape: 1 24 96 128
1.163186
0.742099
0.194119
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410
0.204410

------ 网络结构 ----
`
TestPartModule(
  (feature_1): Sequential(
    (0): ConvBNReLU(
      (0): Conv2d(3, 32, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
      (1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): ReLU6(inplace=True)
    )
    (1): InvertedResidual(
      (conv): Sequential(
        (0): ConvBNReLU(
          (0): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), groups=32, bias=False)
          (1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU6(inplace=True)
        )
        (1): Conv2d(32, 16, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (2): BatchNorm2d(16, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
    (2): InvertedResidual(
      (conv): Sequential(
        (0): ConvBNReLU(
          (0): Conv2d(16, 96, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (1): BatchNorm2d(96, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU6(inplace=True)
        )
        (1): ConvBNReLU(
          (0): Conv2d(96, 96, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), groups=96, bias=False)
          (1): BatchNorm2d(96, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU6(inplace=True)
        )
        (2): Conv2d(96, 24, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (3): BatchNorm2d(24, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
    (3): InvertedResidual(
      (conv): Sequential(
        (0): ConvBNReLU(
          (0): Conv2d(24, 144, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (1): BatchNorm2d(144, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU6(inplace=True)
        )
        (1): ConvBNReLU(
          (0): Conv2d(144, 144, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), groups=144, bias=False)
          (1): BatchNorm2d(144, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU6(inplace=True)
        )
        (2): Conv2d(144, 24, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (3): BatchNorm2d(24, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
  )
)

`

**Comment by jxt1234** [comment]
我们在解决中，请稍候

**Comment by lqian** [comment]
逐个赋值给input host tensor中的元素为1，输出结果和onnx/pytorch一致
`Tensor host_input(inputTensor, inputTensor->getDimensionType());
	auto host_buffer = host_input.host<float>();
	inputTensor->copyFromHostTensor(&host_input);
	net->runSession(session);
	tensor = net->getSessionOutput(session, "output");
	tensor->copyToHostTensor(&hostTensor);
	buffer = hostTensor.host<float>();
		for (int i=0; i < 20; i++) {
			printf("%2.6f\n", buffer[i]);
		}

	printf("done with fill tensor!\n");`

//output is
1.531728
1.195619
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
0.709804
done with fill tensor!

**Comment by lqian** [comment]
使用menset批量赋值，输出结果不一致， 难道host tensor的buffer是不连续的？
memset(host_buffer, 1.0f, sizeof(float) * inputWidth * inputHeight * 3);
//	//1.531728
//	1.195619
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804
//	0.709804

**Comment by jxt1234** [comment]
host_input.host(); 这类代码有问题，话说你应该编译不过吧

**Comment by jxt1234** [comment]
应该是 host<float>()

**Comment by jxt1234** [comment]
使用问题，fixed
